debian/tests/unit-tests

```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+make check || (cat test-suite.log && false)
```

Comment:
we run this at package build time already.

The autopkgtest tests are meant to test the *installed* binaries, not the built ones. They could do stuff like "make sure the init script does what it's supposed to do" etc, but shouldn't run the functional test suite.

I don't think this kind of autopkgtest is of any help, so I'd prefer against it.

Reply:
Ah that's not very well explained here https://lintian.debian.org/tags/testsuite-autopkgtest-missing.html .

Do you mean like checking command line arguments are spelt correctly/to match the code in the init script? :tongue: 

We could do something like this @yoe https://anonscm.debian.org/viewvc/python-apps/packages/vdirsyncer/trunk/debian/tests/control?revision=13541&view=markup or is even that not really worth it? Is it worth putting it in as a starting point.

Reply:
That could work as a first attempt, yes.

What would make somewhat more sense is to have an autopkgtest that starts the daemon and then tries to run some things through command line utilities or the browser (e.g., using selenium or some such) to make sure that once installed, the packages work as designed.

That'd be quite some work to set up though.

Reply:
Yeah, well I guess the ultimate, and something I've always wanted to do on our normal CI would be to run up olad, then run the RDM tests against the dummy responders (or even run the RDM test server, and run the suite via the webpage).

But I've no idea how you describe that sort of inter-package dependency and test requirements, and where you'd script what...

Reply:
That sounds like something awesome to look into, and shouldn't need as much time (since much of the code has already been written, as opposed to the selenium stuff).

Re: package interdependencies: that should be simple, the autopkgtest spec allows you to declare dependencies needed for the test suite. The test system will ensure that the packages are installed before running the tests. Your script may then check that the right daemons are running (or fiddle with them until they are) and that's it.

Reply:
Okay, I've done the boring simple thing for now, based on the one I linked to. We can look at extending the RDM test stuff when we get it working in Travis first I guess.